instrumentation/jaxrs-client-2.0/src/test/java/io/opentelemetry/instrumentation/hypertrace/jaxrs/v2_0/JaxrsClientBodyInstrumentationTest.java

```diff
@@ -0,0 +1,157 @@
+            .get(HypertraceSemanticAttributes.httpResponseHeader("test-response-header")));
+    Assertions.assertEquals(
+        "test-header-value",
+        clientSpan
+            .getAttributes()
+            .get(HypertraceSemanticAttributes.httpRequestHeader("test-request-header")));
+    Assertions.assertNull(
+        clientSpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_REQUEST_BODY));
+    SpanData responseBodySpan = traces.get(0).get(1);
+    Assertions.assertEquals(
+        GetJsonHandler.RESPONSE_BODY,
+        responseBodySpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_RESPONSE_BODY));
+  }
+
+  @Test
+  public void postJson() throws TimeoutException, InterruptedException {
+    ClientBuilder clientBuilder = ClientBuilder.newBuilder();
+    Client client = clientBuilder.build();
+
+    Response response =
+        client
+            .target(String.format("http://localhost:%d/post", testHttpServer.port()))
+            .request()
+            .header("test-request-header", "test-header-value")
+            .post(Entity.entity(JSON, MediaType.APPLICATION_JSON_TYPE));
+    Assertions.assertEquals(204, response.getStatus());
+
+    TEST_WRITER.waitForTraces(1);
+    List<List<SpanData>> traces = TEST_WRITER.getTraces();
+    Assertions.assertEquals(1, traces.size());
+    Assertions.assertEquals(1, traces.get(0).size());
+    SpanData clientSpan = traces.get(0).get(0);
+
+    Assertions.assertEquals(
+        "test-value",
+        clientSpan
+            .getAttributes()
+            .get(HypertraceSemanticAttributes.httpResponseHeader("test-response-header")));
+    Assertions.assertEquals(
+        JSON, clientSpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_REQUEST_BODY));
+    Assertions.assertNull(
+        clientSpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_RESPONSE_BODY));
+  }
+
+  @Test
+  public void postUrlEncoded() throws TimeoutException, InterruptedException {
+    ClientBuilder clientBuilder = ClientBuilder.newBuilder();
+    Client client = clientBuilder.build();
+
+    WebTarget webTarget =
+        client.target(String.format("http://localhost:%d/post", testHttpServer.port()));
+    MultivaluedMap<String, String> formData = new MultivaluedHashMap<>();
+    formData.add("key1", "value1");
+    formData.add("key2", "value2");
+    Response response = webTarget.request().post(Entity.form(formData));
+    Assertions.assertEquals(204, response.getStatus());
+
+    TEST_WRITER.waitForTraces(1);
+    List<List<SpanData>> traces = TEST_WRITER.getTraces();
+    Assertions.assertEquals(1, traces.size());
+    Assertions.assertEquals(1, traces.get(0).size());
+    SpanData clientSpan = traces.get(0).get(0);
+
+    Assertions.assertEquals(
+        "test-value",
+        clientSpan
+            .getAttributes()
+            .get(HypertraceSemanticAttributes.httpResponseHeader("test-response-header")));
+    Assertions.assertEquals(
+        "{key1=[value1], key2=[value2]}",
```

Comment:
@davexroth this is the attribute value for the request body with URL encoded content type.  Should it be captured in some other form e.g. `foo=bar&baz=bar`?

Reply:
changed to `foo=bar&baz=bar` that is supported by the platform.